File: pre-contest/b_mayan_treasure/sol/b_mayan_treasure.java
import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.Scanner;

public class MainShaker {

	public static void main(String[] args) {
		Scanner scanner = new Scanner(System. in);
		int N = scanner.nextInt();
		List<Integer> tab = new ArrayList<>();
		
		for(int i = 0; i < N; i++) {
			tab.add(scanner.nextInt());
		}
		
		Collections.sort(tab);
		
		int a = 0, b = 0, c = 0, mini = -1;
		boolean first = true;
		
		for(int i = 0; i < N - 2; i++) {
			if(first || tab.get(i + 2) - tab.get(i) < mini) {
				first = false;
				a = tab.get(i);
				b = tab.get(i + 1);
				c = tab.get(i + 2);
				mini = tab.get(i + 2) - tab.get(i);
			}
		}
		
		System.out.println(a + " " + b + " " + c);
		
		scanner.close();
	}
}

File: pre-contest/b_mayan_treasure/sol/b_mayan_treasure.cpp
#include "testlib.h"
#include <iostream>
#include <cassert>
#include <vector>
#include <unordered_set>
#include <algorithm>
#include <random>
using namespace std;

int main(int argc, char* argv[])
{
    int N, a  = 0, b = 0, c = 0, mini = -1;
    vector<int> tab;
    bool first = true;

    cin >> N;
    for(int i = 0; i < N; i++){
    	int tmp;
    	cin >> tmp;
    	tab.push_back(tmp);
    }

    sort(tab.begin(), tab.end());

    for(int i = 0; i < N - 2; i++){
    	if(first or tab[i+2]-tab[i] < mini){
	        first = false;
	        a = tab[i];
	        b = tab[i+1];
	        c = tab[i+2];
	        mini = tab[i+2]-tab[i];
	    }
    }

    cout << a << " " << b << " " << c;
}
